File: area.py
import math


class Rectangle:
    def __init__(self, width, height):
        self.width = width
        self.height = height

    def area(self):
        return self.width * self.height

class Circle:
    def __init__(self, radius):
        self.radius = radius

    def area(self):
        return math.pi * self.radius**2

rect = Rectangle(3, 4)
circ = Circle(5)

print(rect.area())  # 12
print(circ.area())  # 78.53981633974483

File: area.jl
abstract type Shape end

struct Rectangle <: Shape
    width::Float64
    height::Float64
end

struct Circle <: Shape
    radius::Float64
end

area(shape::Rectangle) = shape.width * shape.height
area(shape::Circle) = π * shape.radius^2

rect = Rectangle(3, 4)
circ = Circle(5)

println(area(rect))  # 12.0
println(area(circ))  # 78.53981633974483
